packages/functional-tests/src/app.ts

```diff
@@ -83,4 +85,56 @@ export default class App {
             delete this.activeTests[testName];
         }
     }
+
+
+    private async launchTestBrowser() {
+        while (true) {
+            const tester = MRESDK.Actor.CreateEmpty(this.context, {});
+            var selectedTestName = undefined;
+            await new Promise<void>((resolve) => {
+                let y = 0;
+                for (let key in this.testFactories) {
+                    console.log(key);
+                    let button = MRESDK.Actor.CreatePrimitive(this.context, {
+                        definition: {
+                            shape: MRESDK.PrimitiveShape.Box,
+                            dimensions: { x: 0.3, y: 0.3, z: 0.01 }
+                        },
+                        addCollider: true,
+                        actor: {
+                            name: key,
+                            parentId: tester.value.id,
+                            transform: {
+                                position: { x: 0, y: y, z: 0 }
+                            }
+                        }
+                    });
+
+                    const buttonBehavior = button.value.setBehavior(MRESDK.ButtonBehavior);
+                    buttonBehavior.onClick('pressed', (userId: string) => {
+                        selectedTestName = button.value.name;
+                        resolve();
+                    });
+
+                    MRESDK.Actor.CreateEmpty(this.context, {
+                        actor: {
+                            name: 'label',
+                            parentId: tester.value.id,
+                            text: {
+                                contents: key,
+                                height: 0.5,
+                                anchor: MRESDK.TextAnchorLocation.MiddleLeft
+                            },
+                            transform: {
+                                position: { x: 0.5, y: y, z: 0 }
+                            }
+                        }
+                    });
+                    y = y + 0.5;
+                }
+            });
+            destroyActors(tester.value);
+            await this.startTest(selectedTestName);

```

Comment:
There's a subtle issue here. At the end of `startTest`, it sends an rpc message to the testbed host to close the websocket. This will trigger a `userLeft` event so you'll never see the functional test menu again. Try commenting out this call at the bottom of startTest:
`this.rpc.send('functional-test:close-connection');`

Reply:
yes, that fixes the test restarting. thank you